src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -50,6 +71,77 @@ public static Name parseName(String name) throws ParseException {
         return new Name(trimmedName);
     }
 
+    /**
+     * Parses a {@code String company} into a {@code Company}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code company} is invalid.
+     */
+    public static Company parseCompany(String company) throws ParseException {
+        requireNonNull(company);
+        String trimmedCompany = company.trim();
+        if (!Company.isValidCompany(trimmedCompany)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        }
+        return new Company(trimmedCompany);
+    }
+
+    /**
+     * Parses a {@code String jobTitle} into a {@code JobTitle}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code jobTitle} is invalid.
+     */
+    public static JobTitle parseJobTitle(String jobTitle) throws ParseException {
+        requireNonNull(jobTitle);
+        String trimmedJobTitle = jobTitle.trim();
+        if (!JobTitle.isValidJobTitle(trimmedJobTitle)) {
+            throw new ParseException(Name.MESSAGE_CONSTRAINTS);
+        }
+        return new JobTitle(trimmedJobTitle);
+    }
+
+    /**
+     * Parses a {@code String Address} into a {@code Address}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code Address} is invalid.
+     */
+    public static Address parseAddress(String address) throws ParseException {
+        requireNonNull(address);
+        String trimmedAddress = address.trim();
+        if (trimmedAddress.contains("l/")) {
```

Comment:
I think you don't need this if-statement because if you do trimmedAddress.split(" l/ ") and there is no matching substring, it returns a 1-element array of the original string anyway. You can just do a split and validation without checking if the label exists because split() will return what you need regardless.

Reply:
right on. I will go change it